src/main/java/com/debatetimer/scheduler/PollCleanupScheduler.java

```diff
@@ -0,0 +1,27 @@
+package com.debatetimer.scheduler;
+
+import com.debatetimer.domain.poll.PollStatus;
+import com.debatetimer.domainrepository.poll.PollDomainRepository;
+import java.time.LocalDateTime;
+import lombok.RequiredArgsConstructor;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+@RequiredArgsConstructor
+public class PollCleanupScheduler {
+
+    private static final int INTERVAL_HOURS = 12;
+    private static final long INTERVAL_MILLIS = INTERVAL_HOURS * 60 * 60 * 1000L;
+    static final int TIMEOUT_HOURS = 3;
+
+    private final PollDomainRepository pollDomainRepository;
+
+    @Scheduled(fixedRate = INTERVAL_MILLIS)
```

Comment:
timezone 설정해주어도 좋을 것 같아요!

```suggestion
    @Scheduled(fixedRate = INTERVAL_MILLIS,  zone = "Asia/Seoul"))
```

Reply:
추가로, 지금은 스케쥴링 로직이 하나여서 상관없는데 Scheduled는 단일 스레드로 처리되어서 두 개이상의 스케쥴링 로직이 있을 때 에러 가능성이 있을 수 있다고 알고 있어요. Configurer로 명확한 스레드 prefix 와 threadExecutor를 지정해주어도 좋을 것 같습니다.

```java
@Configuration
@EnableScheduling
public class SchedulerConfig implements SchedulingConfigurer {
    @Override
    public void configureTasks(ScheduledTaskRegistrar taskRegistrar) {
        ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
 
        threadPoolTaskScheduler.setPoolSize(10);
        threadPoolTaskScheduler.setThreadNamePrefix("my-scheduler-");
        threadPoolTaskScheduler.initialize();
 
        taskRegistrar.setTaskScheduler(threadPoolTaskScheduler);
    }
}
```

관련 포스팅 : https://dkswnkk.tistory.com/728

Reply:
timezone은 추가했습니다.
그런데 스케줄링 설정들에 대해서 저도 할까말까 고민은 했었지만 지금 이 기능은 위의 것들과 전혀 상관없다는 생각을 했습니다. 일부러 단일 스레드로 처리하라고 명시했고 저희 서비스의 경우 트래픽이 크지 않고 그럴 가능성도 적은 도메인이라 threadExecutor를 따로 구성할 필요가 없다고 생각했습니다. 추후 스케쥴링이 추가적으로 필요하면 그때 도입할 생각이었습니다.